syntax="proto3";
import "common.proto";
package cmq;

message GetBrokerRequest
{
    string topic = 1;
}

message GetBrokerResponse
{
    BrokerInfo broker_info = 1;
    Status status = 2;
}

message RegisterRequest
{
    string ip = 1;
    int32 port = 2;
}

message RegisterResponse
{
    Status status = 1;
}

message UnRegisterRequest
{
    string ip = 1;
    int32 port = 2;
}

message UnRegisterResponse
{
    Status status = 1;
}

service cmq {
    rpc GetBroker (GetBrokerRequest) returns (GetBrokerResponse);
    rpc Register (RegisterRequest) returns (RegisterResponse);
    rpc UnRegister (UnRegisterRequest) returns (UnRegisterResponse);
}
